discovery-marathon-api/src/main/scala/akka/discovery/marathon/MarathonApiSimpleServiceDiscovery.scala

```diff
@@ -84,7 +84,17 @@ class MarathonApiSimpleServiceDiscovery(system: ActorSystem) extends SimpleServi
 
       entity <- response.entity.toStrict(resolveTimeout)
 
-      appList <- Unmarshal(entity).to[AppList]
+      appList <- {
+        system.log.debug("Marathon API entity: [{}]", entity.data.utf8String)
+        val unmarshalled = Unmarshal(entity).to[AppList]
+
+        unmarshalled.failed.foreach { _ =>
+          system.log.error("Failed to unmarshal Marathon API response status [{}]", response.status.value)
+          system.log.error("Marathon API entity: [{}]", entity.data.utf8String)
+          system.log.error("URI: {}", uri)
+        }
```

Comment:
likely better as one log statement, otherwise hard to relate where which URI this error was for as there technically could be multiple logs being issued at the same time; if combined in one log statement this is LGTM then :)